src/FileMigrationHelper.php

```diff
@@ -102,6 +102,17 @@ protected function migrateFile($base, File $file, $legacyFilename)
             return false;
         }
 
+        // If this file has the wrong class for it's extension, update the classname
+        $destinationClass = $file->get_class_for_file_extension($extension);
+        if ($file->ClassName !== $destinationClass) {
+            // We disable validation so that we are able to write a corrected
+            // classname, once that is changed we re-enable it for subsequent writes
+            Config::modify()->set(File::class, 'validation_enabled', false);
+            $file = $file->newClassInstance($destinationClass);
+            $file->write();
```

Comment:
This line is still throwing a `[Emergency] Uncaught SilverStripe\ORM\ValidationException: Extension 'pdf' is not allowed` error for me despite disabling validation.

Reply:
Fixed, it worked if you ran it multiple times which is why I didn't pick it up. It now get's the updated file before continuing.